x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/multivalue/AbstractMultivalueFunctionTestCase.java

```diff
@@ -314,7 +325,64 @@ protected static void longs(
     }
 
     /**
-     * Build a test case with unsigned {@code long} values.
+     * Build many test cases with {@code long} values.
+     */
+    protected static void dateTimes(
+        List<TestCaseSupplier> cases,
+        String name,
+        String evaluatorName,
+        BiFunction<Integer, LongStream, Matcher<Object>> matcher
+    ) {
+        dateTimes(cases, name, evaluatorName, DataTypes.DATETIME, matcher);
+    }
+
+    /**
+     * Build many test cases with {@code date} values.
+     */
+    protected static void dateTimes(
+        List<TestCaseSupplier> cases,
+        String name,
+        String evaluatorName,
+        DataType expectedDataType,
+        BiFunction<Integer, LongStream, Matcher<Object>> matcher
+    ) {
+        cases.add(
+            new TestCaseSupplier(
+                name + "(epoch)",
+                List.of(DataTypes.DATETIME),
+                () -> new TestCaseSupplier.TestCase(
+                    List.of(new TestCaseSupplier.TypedData(List.of(0L), DataTypes.DATETIME, "field")),
+                    evaluatorName + "[field=Attribute[channel=0]]",
+                    expectedDataType,
+                    matcher.apply(1, LongStream.of(0L))
+                )
+            )
+        );
+        cases.add(new TestCaseSupplier(name + "(long)", List.of(DataTypes.DATETIME), () -> {
+            long data = randomMillisUpToYear9999();
```

Comment:
These are all non-negative, so we never test years before 1970. IMO we should test that as well.

Also applies below in line 372.

Reply:
Some stuff doesn't like negative dates sometimes. I'll try it with `randomLong` which'll produce some wild dates, but should cover all the things.